internal/eval/eval_test.go

```diff
@@ -867,6 +868,108 @@ func testEvalJSONSET(t *testing.T, store *dstore.Store) {
 	runEvalTests(t, tests, evalJSONSET, store)
 }
 
+func testEvalJSONARRAPPEND(t *testing.T, store *dstore.Store) {
```

Comment:
Add a test case where you are appending elements of different kinds to the array. Add it into integration tests too.
```sh
0.0.0.0:6379> JSON.SET arr $ '[1,2,3,"blue"]'
OK
0.0.0.0:6379> JSON.ARRAPPEND arr $ '["black"]'
1) (integer) 5
0.0.0.0:6379> JSON.GET arr
"[1,2,3,\"blue\",[\"black\"]]"
0.0.0.0:6379>
```